### ISSUE TITLE
Overrides don't work for local values

### ISSUE THREAD

**Issue by screamingwalls** [hashicorp/terraform#18351]
### Terraform Version

```
Terraform v0.11.7

...
```

### Terraform Configuration Files

```hcl
::::::::::::::
main.tf
::::::::::::::
locals {
    x = 1
}
::::::::::::::
override.tf
::::::::::::::

locals {
    x = 2
}

...
```

### Expected Behavior

Terraform should have reported:
No changes. Infrastructure is up-to-date.

### Actual Behavior

Error: x: duplicate local. local value names must be unique
### Steps to Reproduce

Create the two files listed above in a directory and run terraform apply.  

### Additional Context

Overrides work for normal variables but not for local vars.

**Comment by apparentlymart** [comment]
Hi @screamingwalls! Sorry for this bug, and thanks for reporting it.

The handling of override files has been overhauled in our development branch for the next major release, and I believe this issue was already fixed as part of that work. Development is still in progress for that, so I've labelled this issue in a way that will allow us to easily find it again and verify the fix as part of our release process once the main development phase is complete.

Thanks again for reporting this!

**Comment by nsaud01** [comment]
I'm running into this same issue with TF v0.11.10

**Comment by apparentlymart** [comment]
Hi again @screamingwalls! Sorry for the long silence here.

I've just verified this on the v0.12.0-alpha2 prerelease build using the following configuration files, based on yours...

main.tf:

```hcl
locals {
  x = 1
}

output "x" {
  value = local.x
}
```

override.tf:

```hcl
locals {
  x = 2
}
```

This is now behaving as expected:

```
$ terraform apply

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

x = 2
```

Since the fix for this is already in master and ready to be included in the final v0.12.0 release, I'm going to close this out. Thanks again for reporting this!

**Comment by kenchan0130** [comment]
The same problem seems to be happening with at least v0.12.19 and v0.12.20.

```
$ terraform plan

Error: Duplicate local value definition

  on sample2.tf line 2, in locals:
   2:   sample_variables = "hoge"

A local value named "sample_variables" was already defined at sample1.tf:2,3-28. Local value names must be unique within a module.
```
